src/tests/ftest/util/AgentUtils.py

```diff
@@ -127,31 +127,34 @@ def run_agent(basepath, server_list, client_list=None):
         sessions[client] = p
 
     # double check agent launched successfully
-    file_desc = sessions[client].stderr.fileno()
-    flags = fcntl.fcntl(file_desc, fcntl.F_GETFL)
-    fcntl.fcntl(file_desc, fcntl.F_SETFL, flags | os.O_NONBLOCK)
     timeout = 5
-    start_time = time.time()
-    result = 0
-    pattern = "Starting daos_agent"
-    expected_data = ""
-    while True:
-        output = ""
-        try:
-            output = sessions[client].stderr.read()
-        except IOError as excpn:
-            if excpn.errno != errno.EAGAIN:
-                raise excpn
-            continue
-        match = re.findall(pattern, output)
-        expected_data += output
-        result += len(match)
-        if not output or time.time() - start_time > timeout:
-            print("<AGENT>: {}".format(expected_data))
-            raise AgentFailed("DAOS Agent didn't start!")
-        break
-    print("<AGENT> agent started and took %s seconds to start" % \
-            (time.time() - start_time))
+    for client in client_list:
+        file_desc = sessions[client].stderr.fileno()
+        flags = fcntl.fcntl(file_desc, fcntl.F_GETFL)
+        fcntl.fcntl(file_desc, fcntl.F_SETFL, flags | os.O_NONBLOCK)
+        start_time = time.time()
+        pattern = "Starting daos_agent"
+        expected_data = ""
+        while True:
+            if time.time() - start_time > timeout:
+                print("<AGENT>: {}".format(expected_data))
+                raise AgentFailed("DAOS Agent didn't start!")
+            output = ""
+            try:
+                output = sessions[client].stderr.read()
+            except IOError as excpn:
+                if excpn.errno != errno.EAGAIN:
+                    raise AgentFailed("Error in starting daos_agent: "
```

Comment:
Is it useful to report which client(s) it failed on?  I guess without some restructuring (to store the failed clients in a list and only raise an exception at the end of the loop) one would only get the first client anyway.

Reply:
Without restructuring the logic here, I don't think I could. It'd fail to launch on the first node and immediately bail out. I prefer that behavior versus running through all of them even if one fails, but I could see this maybe being worth a ticket in the future.

Reply:
Agreed that restructuring to get all failed agent starts is probably a different ticket but even reporting the one that raises the `AgentFailed()` here might be useful so that somebody trying to debug the issue at least knows which one caused the run to fail and knows which logs to look into.